app/src/main/java/com/example/codingquiz/ui/screen/QuestionScreen.kt

```diff
@@ -47,11 +50,12 @@ class AnswerState {
 fun QuestionScreen(
     questionViewModel: QuestionViewModel = koinViewModel(),
     givenAnswerViewModel: GivenAnswerViewModel = koinViewModel(),
-    categoryId: Int?,
+    category: Category?,
```

Comment:
NEIN, NEIN, NEIN.
```suggestion
    category: Category,
```
Check nullability in the upper layer and construct this screen only if the deserialized category won't be null.

Reply:
Fixed :)